File: Examples/Sample-Objc-UISceneDelegate/Sample-Objc-UISceneDelegate/ViewController.m
//
//  ViewController.m
//  Sample-Objc-UISceneDelegate
//
//  Created by Michael McNamara on 03/10/2019.
//  Copyright © 2019 Intercom. All rights reserved.
//

#import "ViewController.h"
#import <Intercom/Intercom.h>

@interface ViewController () <UIAlertViewDelegate>

@property (nonatomic, weak) IBOutlet UIButton *logoutButton;
@property (nonatomic, weak) IBOutlet UIButton *loginButton;
@property (nonatomic, weak) IBOutlet UIButton *sendMessageButton;
@property (nonatomic, assign) BOOL loggedIn;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.loggedIn = [[NSUserDefaults standardUserDefaults] objectForKey:@"email"] != nil;
}

- (void)setLoggedIn:(BOOL)loggedIn {
    _loggedIn = loggedIn;
    
    self.loginButton.enabled = !loggedIn;
    self.logoutButton.enabled = loggedIn;
    self.sendMessageButton.enabled = loggedIn;
}

- (IBAction)loginPressed:(id)sender {
    //The user is not logged in, so prompt for their email address
    
    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"User Login" message:@"Type in an email address" preferredStyle:UIAlertControllerStyleAlert];
    [alertController addAction:[UIAlertAction actionWithTitle:@"Cancel" style:UIAlertActionStyleCancel handler:nil]];
    [alertController addAction:[UIAlertAction actionWithTitle:@"Login" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        [self handleUserLogin:alertController];
    }]];
    [alertController addTextFieldWithConfigurationHandler:^(UITextField * _Nonnull textField) {
        textField.placeholder = @"Email";
    }];
    [self presentViewController:alertController animated:YES completion:nil];
}

- (IBAction)logoutPressed:(id)sender {
    //Logout was pressed, so calling [Intercom logout] will log remove all local user data and stop tracking them.
    [Intercom logout];
    
    //Save email so we know the user is logged in
    [[NSUserDefaults standardUserDefaults] removeObjectForKey:@"email"];
    [[NSUserDefaults standardUserDefaults] synchronize];
    
    self.loggedIn = NO;
}

- (IBAction)openIntercomPressed:(id)sender {
    [Intercom presentMessenger];
}

- (void)handleUserLogin:(UIAlertController *)alertController {
    //The user pressed login, if they gave us an email, log them in.
    UITextField *emailField = alertController.textFields.firstObject;
    NSString *email = emailField.text;
    if (email.length > 0) {
        //Start tracking the user with Intercom
        [Intercom registerUserWithEmail:email];
        
        //Save email so we know the user is logged in
        [[NSUserDefaults standardUserDefaults] setObject:email forKey:@"email"];
        [[NSUserDefaults standardUserDefaults] synchronize];
        
        self.loggedIn = YES;
    }
}

@end

File: Sample-Swift/Sample-Swift/ViewController.swift
//
//  ViewController.swift
//  Sample-Swift
//
//  Created by Brian Boyle on 20/07/2016.
//  Copyright © 2016 Intercom. All rights reserved.
//

import UIKit
import Intercom

let emailKey = "email"

class ViewController: UIViewController {

    @IBOutlet weak var loginButton: UIButton!
    @IBOutlet weak var logoutButton: UIButton!
    @IBOutlet weak var openIntercomButton: UIButton!
    var loggedIn: Bool! = false {
        didSet {
            loginButton.enabled = !loggedIn
            logoutButton.enabled = loggedIn
            openIntercomButton.enabled = loggedIn
        }
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        loggedIn = NSUserDefaults.standardUserDefaults().stringForKey(emailKey) != nil
    }
    
    
    @IBAction func loginTapped(sender: UIButton) {
        //The user is not logged in, so prompt for their email address
        
        let loginAlert = UIAlertController(title: "User Login", message: "Type in an email address", preferredStyle: UIAlertControllerStyle.Alert)
        loginAlert.addAction(UIAlertAction(title: "Cancel", style: UIAlertActionStyle.Cancel, handler: nil))
        loginAlert.addAction(UIAlertAction(title: "Login", style: UIAlertActionStyle.Default, handler: { (action: UIAlertAction) in
            self.handleLogin(loginAlert)
        }))
        loginAlert.addTextFieldWithConfigurationHandler { (textField : UITextField) in
            textField.placeholder = "Enter email"
        }
        loginAlert.view.setNeedsLayout()
        self.presentViewController(loginAlert, animated: true, completion: nil)
    }


    @IBAction func logoutTapped(sender: UIButton) {
        //Logout was pressed, so calling [Intercom reset] will log remove all local user data and stop tracking them.
        Intercom.reset()
        NSUserDefaults.standardUserDefaults().removeObjectForKey(emailKey)
        loggedIn = false
    }
    
    
    @IBAction func openIntercomTapped(sender: UIButton) {
        Intercom.presentMessenger()
    }
    
    func handleLogin(alertController: UIAlertController) -> Void {
        
        let textField = (alertController.textFields?.first)!
        if !(textField.text?.isEmpty)! {
            //Start tracking the user with Intercom
            Intercom.registerUserWithEmail(textField.text!)
            
            NSUserDefaults.standardUserDefaults().setObject(textField.text!, forKey: emailKey)
            loggedIn = true
        }
    }
}

